boards/shared/Src/Io/Io_SharedFreqOnlyPwmInput.c

```diff
@@ -0,0 +1,133 @@
+#include "main.h"
+#include <math.h>
+#include <assert.h>
+#include "Io_SharedFreqOnlyPwmInput.h"
+
+#define MAX_TIMER_PERIOD_ELAPSED_COUNT 2
+
+struct FreqOnlyPwmInput
+{
+    float              frequency_hz;
+    TIM_HandleTypeDef *htim;
+    float              timer_frequency_hz;
+    uint32_t           rising_edge_tim_channel;
+    uint32_t           timer_auto_reload_reg;
+    bool               first_tick;
+    uint32_t           curr_rising_edge;
+    uint32_t           prev_rising_edge;
+
+    // Elapsed tick counts the number of times which a timer overflow occurs
+    size_t elapsed_tick;
+};
+
+/**
+ * Set the frequency for the given PWM input
+ * @param pwm_input: The PWM input to set frequency for
+ * @param frequency_hz: Frequency, in Hz
+ */
+static void Io_SetFrequency(
+    struct FreqOnlyPwmInput *pwm_input,
+    const float              frequency_hz);
+
+static void Io_SetFrequency(
+    struct FreqOnlyPwmInput *const pwm_input,
+    const float                    frequency_hz)
+{
+    assert(frequency_hz >= 0.0f || isnanf(frequency_hz));
```

Comment:
This logic is wrong it should be `&&`

Reply:
We should expect a float GT `0.0` or `NaN` as inputs to this function. If we made it `&&` it would break because positive floating point numbers aren't considered as `NaN`.

Reply:
Oh i misread the intention